### PR TITLE
[MM-30539] License renewal link endpoint

### PR DESCRIPTION
**Pull request by ethervoid**
#### Summary

Doc for the new license renewal link endpoint

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-30539

### CODE REVIEW

File: v4/source/system.yaml [hunk 1]
```diff
@@ -737,6 +737,34 @@
...
+        - system
+      summary: Request the license renewal link
+      description: >
+        Request the renewal link that would be used to start the license renewal process
+
+        __Minimum server version__: 5.32
+
+        ##### Permissions
+
+        Must have `manage_system` permission.
```
Comment (marianunez) [inline v4/source/system.yaml:752]: Is this accurate? It seems we are checking the read about license permission. Should we fallback in this one instead?

Comment (ethervoid) [reply v4/source/system.yaml]: I've updated that to reflect the permission needed

**Review by marianunez — APPROVED** [review]

**Comment by agnivade** [comment]
@ethervoid - can we merge this? This is breaking master because the related PR got merged.

**Comment by ethervoid** [comment]
@agnivade merged! sorry!

**Pull request merged**